pkg/controllers/disruption/consolidation_test.go

```diff
@@ -749,507 +827,749 @@ var _ = Describe("Consolidation", func() {
 					},
-				},
-				ResourceRequirements: v1.ResourceRequirements{
-					Requests: v1.ResourceList{
-						v1.ResourceCPU: resource.MustParse("2"),
-					},
-				},
-			})
-			nodeClaim2, node2 := test.NodeClaimAndNode(v1beta1.NodeClaim{
-				ObjectMeta: metav1.ObjectMeta{
-					Labels: map[string]string{
-						v1beta1.NodePoolLabelKey:     nodePool.Name,
-						v1.LabelInstanceTypeStable:   mostExpensiveInstance.Name,
-						v1beta1.CapacityTypeLabelKey: mostExpensiveOffering.CapacityType,
-						v1.LabelTopologyZone:         mostExpensiveOffering.Zone,
+					ResourceRequirements: v1.ResourceRequirements{
+						Requests: v1.ResourceList{
+							v1.ResourceCPU: resource.MustParse("2"),
+						},
 					},
-				},
-				Status: v1beta1.NodeClaimStatus{
-					ProviderID: test.RandomProviderID(),
-					Allocatable: map[v1.ResourceName]resource.Quantity{
-						v1.ResourceCPU:  resource.MustParse("5"),
-						v1.ResourcePods: resource.MustParse("100"),
+				})
+				annotatedNodeClaim, annotatedNode := test.NodeClaimAndNode(v1beta1.NodeClaim{
+					ObjectMeta: metav1.ObjectMeta{
+						Annotations: map[string]string{
+							v1beta1.DoNotDisruptAnnotationKey: "true",
+						},
+						Labels: map[string]string{
+							v1beta1.NodePoolLabelKey:     nodePool.Name,
+							v1.LabelInstanceTypeStable:   mostExpensiveInstance.Name,
+							v1beta1.CapacityTypeLabelKey: mostExpensiveOffering.CapacityType,
+							v1.LabelTopologyZone:         mostExpensiveOffering.Zone,
+						},
 					},
-				},
-			})
-			// Block this pod from being disrupted with karpenter.sh/do-not-disrupt
-			pods[2].Annotations = lo.Assign(pods[2].Annotations, map[string]string{v1beta1.DoNotDisruptAnnotationKey: "true"})
+					Status: v1beta1.NodeClaimStatus{
+						ProviderID: test.RandomProviderID(),
+						Allocatable: map[v1.ResourceName]resource.Quantity{
+							v1.ResourceCPU:  resource.MustParse("5"),
+							v1.ResourcePods: resource.MustParse("100"),
+						},
+					},
+				})
 
-			ExpectApplied(ctx, env.Client, rs, pods[0], pods[1], pods[2], nodePool)
-			ExpectApplied(ctx, env.Client, nodeClaim, node, nodeClaim2, node2)
+				if isSpotNode {
+					annotatedNodeClaim, annotatedNode = test.NodeClaimAndNode(v1beta1.NodeClaim{
+						ObjectMeta: metav1.ObjectMeta{
+							Annotations: map[string]string{
+								v1beta1.DoNotDisruptAnnotationKey: "true",
+							},
+							Labels: map[string]string{
+								v1beta1.NodePoolLabelKey:     nodePool.Name,
+								v1.LabelInstanceTypeStable:   mostExpensiveSpotInstance.Name,
+								v1beta1.CapacityTypeLabelKey: mostExpensiveSpotOffering.CapacityType,
+								v1.LabelTopologyZone:         mostExpensiveSpotOffering.Zone,
+							},
+						},
+						Status: v1beta1.NodeClaimStatus{
+							ProviderID: test.RandomProviderID(),
+							Allocatable: map[v1.ResourceName]resource.Quantity{
+								v1.ResourceCPU:  resource.MustParse("5"),
+								v1.ResourcePods: resource.MustParse("100"),
+							},
+						},
+					})
+				}
```

Comment:
I think you could save on some lines here by doing the lo.Assign() as well. There are other places I've not commented this that also apply, but figured you know where Im' talking about.

Reply:
Hmm, not necessarily. I think it is going to add a lot more redundant code if we do this since we will need lo.Assign() for each of the property in NodeClaim and Node causing a lot more boiler plate code.

For example, in this case we will lo.Assign() for Annotations, Labels, Status just for NodeClaim and repeat that for Node too, right ? I have changed this in few places wherever necessary.

Reply:
What did we decide on here? You only need to do lo.Assign() for the labels, since the status and annotations can remain the same.

Reply:
Refactored almost the entire file and used as many lo.Assign() as possible.